engine/configurator/conf_cyaml.h

```diff
@@ -0,0 +1,806 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/** @file
+ * @brief YAML configuration types and schema
+ *
+ * API definitions.
+ *
+ * Copyright (C) 2018-2022 OKTET Labs Ltd. All rights reserved.
+ */
+
+#ifndef __TE_CONF_CYAML_H__
+#define __TE_CONF_CYAML_H__
+
+#include <cyaml/cyaml.h>
+#include "conf_api.h"
+#include "conf_types.h"
+#include "logger_api.h"
+
+/******************************************************************************
+ * C data structure for storing a backup.
+ *
+ * This is what we want to load the YAML into.
+ ******************************************************************************/
+
+/* enumeration for object or instance. It answers if it is object */
+enum scope {
+    CFG_INSTANCE =0, /* It is a default value */
+    CFG_OBJECT
+};
+
+/* Structure for storing depends entry */
+typedef struct depends_entry {
+    char *oid;
+    uint8_t scope; /* optional; 0 is default value means instance */
+} depends_entry;
+
+/* Structure for storing object */
+typedef struct object_type {
+    char *d; /* cyaml doesn'know about this type of field */
```

Comment:
I meant ` char *d; /* cyaml doesn'know about this type of field */`

Reply:
It is a YAML key (it means `d`escription however it doesn't matter). I've added some words in comment to object_type about the fact that the name of the fields coincides with  YAML keys.

Reply:
I still does not understand, how the comment explains the field.

Reply:
This comment doesn't explain the field. Because I think the explanation of the field even doesn't needed.
"The name for the field coincides with the corresponding YAML key ..."

However it is worth to mention that cyaml doen't know about this field. That means CYAML can't put there any information, take it from there, free, or something.